probes/http/http.go

```diff
@@ -129,10 +131,23 @@ func (p *Probe) Init(name string, opts *options.Options) error {
 
 	// Needs to be non-nil so we can set parameters on it.
 	transport := http.DefaultTransport
-
 	// Keep idle connections open until we explicitly close them.
 	// This allows us to send multiple requests over the same connection.
 	transport.(*http.Transport).MaxIdleConnsPerHost = 1
+	if p.protocol == "https" {
+		tlsConfig := &tls.Config{}
```

Comment:
tlsConfig should be part of the probe struct and should be created during probe initialization, instead of the probing path.